### PR TITLE
Auth/context

### PR DESCRIPTION
**Pull request by BhavyaSheth22**
Added login flow, integration, context to manage user state and registration form modifications.

TODO:
1. Persist user state by setting it in localStorage
2. Add registration integration.

### CODE REVIEW

**Comment by BhavyaSheth22** [comment]
User persistence has been added. I'll tackle registration in a different PR

File: client/src/components/LoginRegister/Registration.jsx [hunk 1]
```diff
@@ -95,7 +95,7 @@ const Registration = () => {
     event.preventDefault();
     validateNumber();
     validateName();
-    validateID();
+    // validateID();
```
Comment (ShubhankarKG) [inline client/src/components/LoginRegister/Registration.jsx:98]: Please remove the comments if not required.

File: client/src/components/LoginRegister/Registration.jsx [hunk 2]
```diff
@@ -114,27 +114,37 @@ const Registration = () => {
     if (name === "number") {
       setNumber(value);
     }
-    if (name === "id") {
-      setID(value);
-    }
+    // if (name === "id") {
+    //   setID(value);
+    // }
```
Comment (ShubhankarKG) [inline client/src/components/LoginRegister/Registration.jsx:119]: Please remove the comments if not required.

File: client/src/components/LoginRegister/Registration.jsx [hunk 3]
```diff
@@ -217,7 +215,7 @@ const Registration = () => {
               justifyContent: "space-around"
             }}
           >
-            <FormControl
+            {/* <FormControl
```
Comment (ShubhankarKG) [inline client/src/components/LoginRegister/Registration.jsx:218]: Please remove the comments if not required.

File: client/src/components/LoginRegister/Registration.jsx [hunk 4]
```diff
@@ -233,6 +231,52 @@ const Registration = () => {
                 aria-describedby="my-helper-text"
               />
               <small style={{ color: "red" }}>{warningSuggested}</small>
+            </FormControl> */}
```
Comment (ShubhankarKG) [inline client/src/components/LoginRegister/Registration.jsx:234]: Please remove the comments if not required.

File: client/src/constants.js [hunk 5]
```diff
@@ -1 +1 @@
-export const BASE_URL = "http://localhost:8000/api/";
+export const BASE_URL = "http://localhost:8000/api";
```
Comment (ShubhankarKG) [inline client/src/constants.js:1]: This change will affect some pages in the DashboardLayout folder and Add/ Edit product form. Please make the relevant changes in the necessary files.

File: client/src/components/LoginRegister/inputOTP.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+//TODO: Add styles to page
+
+const OtpPage = (props) => {
+  const dispatch = useAuthDispatch();
+  const history = useHistory();
+  const { loading, errorMessage } = useAuthState();
+
+  const [otp, setOtp] = useState("");
+  // const classes = useStyles();
```
Comment (ShubhankarKG) [inline client/src/components/LoginRegister/inputOTP.jsx:21]: Please remove the comments if not required.

File: client/src/context/reducer.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+export const AuthReducer = (initialState, action) => {
+  switch (action.type) {
+    case "REQUEST_LOGIN":
+      return {
+        ...initialState,
+        loading: true
+      };
+    case "LOGIN_SUCCESS":
+      // TODO: Add userdetails to local storage
```
Comment (ShubhankarKG) [inline client/src/context/reducer.js:23]: Please remove the comments if not required.

File: client/src/context/reducer.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
+let user = localStorage.getItem("user")
+  ? JSON.parse(localStorage.getItem("user"))
+  : "";
+let token = localStorage.getItem("token");
+let isAuthenticated = localStorage.getItem("isAuthenticated") ? true : false;
+
+export const initialState = {
+  user: "" || user,
```
Comment (ShubhankarKG) [inline client/src/context/reducer.js:8]: if the above condition is false, user is by default to an empty string, we can safely set this to only user.

File: client/src/context/reducer.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  token: "" || token,
```
Comment (ShubhankarKG) [inline client/src/context/reducer.js:9]: if the above condition is false, token is by default to an empty string, we can safely set this to only token.

**Review by ShubhankarKG — APPROVED** [review]

**Pull request merged**